test/hotspot/jtreg/serviceability/jvmti/vthread/ToggleNotifyJvmtiTest/ToggleNotifyJvmtiTest.java

```diff
@@ -0,0 +1,178 @@
+            }
+            i = i + 1;
+        }
+    }
+
+    // ensure thread is ready
+    public void ensureReady() {
+        try {
+            while (!threadReady) {
+                sleep(1);
+            }
+        } catch (InterruptedException e) {
+            throw new RuntimeException("Interruption while preparing tested thread: \n\t" + e);
+        }
+    }
+
+    public void letFinish() {
+        shouldFinish = true;
+    }
+}
+
+public class ToggleNotifyJvmtiTest {
+    private static final int VTHREADS_CNT = 60;
+    private static final String AGENT_LIB = "ToggleNotifyJvmtiTest";
+    private static final WhiteBox WB = WhiteBox.getWhiteBox();
+
+    private static native boolean IsAgentStarted();
+    private static native int VirtualThreadStartedCount();
+
+    static void log(String str) { System.out.println(str); }
+
+    static public void sleep(long millis) {
+        try {
+            Thread.sleep(millis);
+        } catch (InterruptedException e) {
+            throw new RuntimeException("Interruption in TestedThread.sleep: \n\t" + e);
+        }
+    }
+
+    static TestedThread[] threads = new TestedThread[VTHREADS_CNT];
+    static Thread vts[] = new Thread[VTHREADS_CNT];
+
+    static private synchronized void startThread(int i) {
+        String name = "TestedThread" + i;
+        TestedThread thread = new TestedThread(name);
+        vts[i] = Thread.ofVirtual().name(name).start(thread);
+        thread.ensureReady();
+        threads[i] = thread;
+        log("## Java: started vthread: " + name);
+    }
+
+    static private void startThreads() {
+        log("\n## Java: Starting vthreads");
+        for (int i = 0; i < VTHREADS_CNT; i++) {
+            startThread(i);
+        }
+    }
+
+    static private synchronized void finishThreads() {
+        log("\n## Java: runIt: Finishing vthreads");
+        try {
+            for (int i = 0; i < VTHREADS_CNT; i++) {
+                TestedThread thread = threads[i];
+                if (thread == null) {
+                    break;
+                }
+                thread.letFinish();
+                vts[i].join();
+            }
+        } catch (InterruptedException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    static private void setVirtualThreadsNotifyJvmtiMode(int iter, boolean enable) {
+        sleep(5);
+        WB.setVirtualThreadsNotifyJvmtiMode(enable);
+        log("# main: SetNotifyJvmtiEvents: #" + iter + " enable: " + enable);
+    }
+
+    public static void main(String[] args) throws Exception {
+        log("# main: loading " + AGENT_LIB + " lib");
+
+        if (args.length > 0 && args[0].equals("attach")) { // agent loaded into running VM case
+            String arg = args.length == 2 ? args[1] : "";
+            VirtualMachine vm = VirtualMachine.attach(String.valueOf(ProcessHandle.current().pid()));
+            vm.loadAgentLibrary(AGENT_LIB, arg);
+        } else {
+            System.loadLibrary(AGENT_LIB);
```

Comment:
Why is this needed? Isn't the library already loaded due to it being specified by `-agentlib`?

Reply:
Good question. We almost always do it in the JVMTI tests including `serviceability/jvmti/vthread` and `vmTestbase/nsk/jvmti` tests. Examples are 22 `serviceability/jvmti/vthread` tests.

Reply:
Are you saying it's not needed, but you included it to be consistent with other tests?

Reply:
I remember it was really needed some time ago but do not remember why.
Removed it now. Then we have a technical debt to get rid of this in all JVMTI tests with the `-agentlib` option.